internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/producer/ApacheKafkaProducerConfig.java

```diff
@@ -253,7 +238,45 @@ public static void copyKafkaSASLProperties(Properties configuration, Properties
     }
   }
 
-  public PubSubMessageSerializer getPubSubMessageSerializer() {
+  protected PubSubMessageSerializer getPubSubMessageSerializer() {
     return pubSubMessageSerializer;
   }
+
+  /**
+   * Validates that the {@link PubSubPositionTypeRegistry} includes support for the Kafka offset-based position type.
+   * <p>
+   * Since this client uses a <b>reserved type ID</b> for Kafka positions (see
+   * {@link PubSubPositionTypeRegistry#APACHE_KAFKA_OFFSET_POSITION_TYPE_ID}), we do not rely on the type ID returned
+   * by the registry for functional behavior. Instead, we perform a <b>sanity check</b> to ensure that the registry
+   * mapping is consistent with the reserved value. This helps detect misconfigured or corrupted mappings.
+   * <p>
+   * If the position class is not registered or the associated type ID does not match the expected reserved value,
+   * this method logs a descriptive error and throws a {@link VeniceException}.
+   *
+   * @param typeRegistry the position type registry to validate
+   * @throws VeniceException if the type is missing or mismatched
+   */
+  private static void validateKafkaPositionType(PubSubPositionTypeRegistry typeRegistry) {
+    if (!typeRegistry.containsFactoryClass(KAFKA_POSITION_FACTORY_CLASS_NAME)) {
```

Comment:
Meaning we will always do strict config check? Maybe it is good, but I wonder do you know when we set strict check to false in the past?